src/bootloaders/shim-systemd.c

```diff
@@ -101,6 +103,10 @@ static char *shim_dst;
 static char *systemd_src;
 static char *systemd_dst;
 
+static char *shim_systemd_get_kernel_dst(const BootManager *manager) {
+        return strdup(KERNEL_DST_DIR);
```

Comment:
Is there a reason why systemd and shim implementation can't both use the destination? (basically just thinking this logic could be the same function between sd_class types and shim).